src/main/kotlin/org/wycliffeassociates/otter/jvm/app/widgets/highlightablebutton/HighlightableButton.kt

```diff
@@ -0,0 +1,53 @@
+package org.wycliffeassociates.otter.jvm.app.widgets.highlightablebutton
+
+import com.jfoenix.controls.JFXButton
+import de.jensd.fx.glyphs.materialicons.MaterialIconView
+import javafx.beans.property.SimpleBooleanProperty
+import javafx.scene.paint.Color
+import tornadofx.*
+
+class HighlightableButton(
+        private val primaryColor: Color,
+        private val secondaryColor: Color,
+        private val hasBorder: Boolean,
+        private val isHighlighted: SimpleBooleanProperty,
+        private val icon: MaterialIconView? = null
+) : JFXButton() {
+
+    init {
+        importStylesheet<HighlightableButtonStyles>()
+
+        addClass(HighlightableButtonStyles.hButton)
+        isDisableVisualFocus = true
+        graphic = icon
+
+        applyColors()
+        isHighlighted.onChange {
+            applyColors()
+        }
+        hoverProperty().onChange {
+            applyColors()
+        }
+    }
+
+    private fun applyColors() {
+        if (hoverProperty().get() || isHighlighted.get()) {
+            applyColors(primaryColor, secondaryColor)
+        } else {
+            applyColors(secondaryColor, primaryColor)
+        }
+    }
+
+    private fun applyColors(bgColor: Color, contentFillColor: Color) {
+        style {
+            backgroundColor += bgColor
+            if(hasBorder) {
+                borderColor += box(primaryColor)
```

Comment:
I believe the borderColor should always be set to the primary color if there is a border, because setting the border color to white causes issues with the background. Also, the border color cannot simply be set once when the button is instantiated, because each time `style` is set here, previous styles get wiped out. Thus, border color must be set here in the parent class, every time `applyColors` is called.

Reply:
what do you think about giving these function different names?  so that they're both not `applyColors()`

Reply:
Sure, I'd probably name the private one something like `doApplyColors()` then

Reply:
what is the += operator doing exactly? are you adding colors? are you appending colors to a list? is there a reason it's not just =?

Reply:
The reason we have to use the `+=` is because the javafx css is expecting a type of  `MutliValue<Paint>` for the backgroundColor. If you jsut use a `=` you get a Type Mismatch